services/webnn/ort/graph_builder_ort.cc

```diff
@@ -629,6 +666,77 @@ void GraphBuilderOrt::AddReshapeOperation(const mojom::Reshape& reshape) {
   CHECK_STATUS(GetOrtGraphApi()->AddNode(graph_.get_ptr(), node.get_pptr()));
 }
 
+void GraphBuilderOrt::AddSliceOperation(const mojom::Slice& slice) {
+  const std::string node_name = GetNodeName(slice.label);
+  const std::string input_name = GetOperandName(slice.input_operand_id);
+  const std::string output_name = GetOperandName(slice.output_operand_id);
+
+  auto range = slice.ranges;
+  base::FixedArray<int32_t> beginnings(slice.ranges.size());
+  base::FixedArray<int32_t> endings(slice.ranges.size());
+  base::FixedArray<int32_t> strides(slice.ranges.size());
+  for (size_t i = 0; i < slice.ranges.size(); ++i) {
+    beginnings[i] = base::checked_cast<int32_t>(slice.ranges[i].start);
+    endings[i] = base::checked_cast<int32_t>(slice.ranges[i].start +
+                                             slice.ranges[i].size);
+    strides[i] = base::checked_cast<int32_t>(slice.ranges[i].stride);
+  }
+
+  // Starts is an operand with data type int64, not an attribute.
+  std::vector<uint32_t> starts_shape = {
+      base::checked_cast<uint32_t>(beginnings.size())};
+  std::vector<int64_t> starts;
+  base::ranges::transform(
+      beginnings, std::back_inserter(starts),
+      [](uint32_t val) { return static_cast<int64_t>(val); });
+  uint64_t starts_id = NewInitializerAsRawData(
```

Comment:
Same here: add a comment explaining why this should be initialized as raw data.

Reply:
Done.